### ISSUE TITLE
Cloud opens in same Browser-Tab

### ISSUE THREAD

**Issue by lemuba** [WLANThermo-nano/WLANThermo_ESP32_Software#69]
Would be nice to have following options as settings:

- same Tab
- new Tab
- new Window

**Comment by sochs** [comment]
Hast du die Optionen nicht auch, wenn du einen rechten Mausklick auf den Link machst?

**Comment by tuniii** [comment]
Ich wäre für einen neuen Tab beim Öffnen der Cloud.

**Comment by sochs** [comment]
Flo, irgendwas war hier doch, dass der Aufruf der neuen Seite nicht möglich war?

**Comment by tuniii** [comment]
Man kann nicht unterscheiden zwischen Tab und Fenster. Das entscheidet der Browser bzw. die Einstellungen im Browser. Ein Tab ist im Prinzip nichts anderes als ein Fenster. Man kann aber unterscheiden, ob ein neues Fenster/Tab oder im gleichen Fenster/Tab geöffnet wird.

**Comment by lemuba** [comment]
> 
> 
> Hast du die Optionen nicht auch, wenn du einen rechten Mausklick auf den Link machst?

Nein, rechter Mausbutton hat dahin gehend keine Funktion/gibt keine Auswahl.

**Comment by arminth** [comment]
Ist kein Link sonderen Skript. Deshalb kein rechter Mausklick.

**Comment by sochs** [comment]
Also auf den Link unter "IoT" kann man durchaus einen rechten Mausklick durchführen. Und am Smartphone funktioniert ein langer Touchdruck um das Kontextmenü des Browsers aufzurufen. Siehe Screenshots:

![AufrufLink](https://user-images.githubusercontent.com/18032460/76859762-36906700-685a-11ea-9958-dc00c69f77b0.png)

![Screenshot_20200317-141637](https://user-images.githubusercontent.com/18032460/76859777-3f813880-685a-11ea-8bff-1bf528493299.png)

**Comment by sochs** [comment]
Wo das nicht geht ist das Wolke-Icon in der Titelleiste, das ist richtig.

**Comment by lemuba** [comment]
Hatte ich auch gerade rausgefunden :-) - aber dann spiegelt doch auf dem Grüne Wolke Symbol einfach den https Link, so das man auch da die Rechte Maustaste/das Kontextmenü anwenden kann. Dann spart man sich erst in IoT reingehen zu müssen.

**Comment by sochs** [comment]
Und genau da bin ich nicht sicher, ich glaube, das ging (bisher) irgendwie nicht. @Phantomias2006

**Comment by Phantomias2006** [comment]
Richtig, das ist nicht so einfach möglich. Warum kann ich auch gerne erklären:
Es gibt 2 "Haupt-API's" data und settings. data wird alle x Sekunden für die Temp-Werte geladen und settings nur bei Bedarf.
Wenn du jetzt auf IOT im Menü klickst dann wird die settings API aufgerufen und die Daten des Menüs befüllt. Somit kann ich beim öffnen der Seite ein normales a-tag verwenden.
Im Header geht hier der Token nicht da dieser in der data API nicht vorhanden ist und ich kein event habe. Deshalb wird beim click auf das Cloud Icon die settings API aufgerufen und die Seite dann mittels JS geöffnet.

Also, wenn das wirklich notwendig ist dann muss der token in die data api.
BG

**Comment by sochs** [comment]
Nicht nur der Token, die ganze URL. Und das würde ich in /data gern vermeiden, die gehört da nicht rein.

**Comment by tuniii** [comment]
Wir könnten "Server Sent Event" einführen. Am Backend können wir dann ein Event setzen und das Frontend aktualisiert sich damit.

**Comment by Phantomias2006** [comment]
Könnte die Lösung sein... Besprechen wir in Slack weiter.
